packages/mgt/src/utils/Cache.ts

```diff
@@ -0,0 +1,361 @@
+   *
+   * @type {number}
+   * @memberof CacheConfig
+   */
+  defaultInvalidationPeriod: number;
+  /**
+   * Cache options for people store
+   *
+   * @type {CacheOptions}
+   * @memberof CacheConfig
+   */
+  people: CacheOptions;
+  /**
+   * Cache options for users store
+   *
+   * @type {CacheOptions}
+   * @memberof CacheConfig
+   */
+  users: CacheOptions;
+  /**
+   * Cache options for photos store
+   *
+   * @type {CacheOptions}
+   * @memberof CacheConfig
+   */
+  photos: CacheOptions;
+}
+
+/**
+ * Options for each store
+ *
+ * @export
+ * @interface CacheOptions
+ */
+export interface CacheOptions {
+  /**
+   * Defines the time (in ms) for objects in the store to expire
+   *
+   * @type {number}
+   * @memberof CacheOptions
+   */
+  invalidiationPeriod: number;
+  /**
+   * Whether the store is enabled or not
+   *
+   * @type {boolean}
+   * @memberof CacheOptions
+   */
+  isEnabled: boolean;
+}
+
+/**
+ * class in charge of managing all the caches and their stores
+ *
+ * @export
+ * @class CacheService
+ */
+// tslint:disable-next-line: max-classes-per-file
+export class CacheService {
+  /**
+   *  Looks for existing cache, otherwise creates a new one
+   *
+   * @static
+   * @template T
+   * @param {CacheSchema} schema
+   * @param {string} store
+   * @returns {Cache<T>}
+   * @memberof CacheService
+   */
+  public static getCache<T extends CacheItem>(schema: CacheSchema, store: string): Cache<T> {
+    const key = `${schema.name}/${store}`;
+
+    if (!this.isInitialized) {
+      this.init();
+    }
+
+    if (!this.cacheStore.has(store)) {
+      this.cacheStore.set(key, new Cache<T>(schema, store));
+    }
+    return this.cacheStore.get(key) as Cache<T>;
+  }
+
+  /**
+   * Globally enables the cache
+   *
+   * @static
+   * @memberof CacheService
+   */
+  public static enableCacheGlobal() {
+    CacheService._cacheConfig = {
+      defaultInvalidationPeriod: 3600000,
+      people: {
+        invalidiationPeriod: CacheService.config.people.invalidiationPeriod,
+        isEnabled: true
+      },
+      photos: {
+        invalidiationPeriod: CacheService.config.photos.invalidiationPeriod,
+        isEnabled: true
+      },
+      users: {
+        invalidiationPeriod: CacheService.config.users.invalidiationPeriod,
+        isEnabled: true
+      }
+    };
+  }
+
+  /**
+   * Globally disables the cache
+   *
+   * @static
+   * @memberof CacheService
+   */
+  public static disableCacheGlobal() {
+    CacheService._cacheConfig = {
+      defaultInvalidationPeriod: 3600000,
+      people: {
+        invalidiationPeriod: CacheService.config.people.invalidiationPeriod,
+        isEnabled: false
+      },
+      photos: {
+        invalidiationPeriod: CacheService.config.photos.invalidiationPeriod,
+        isEnabled: false
+      },
+      users: {
+        invalidiationPeriod: CacheService.config.users.invalidiationPeriod,
+        isEnabled: false
+      }
+    };
+  }
```

Comment:
You can replace this with a single boolean and avoid this duplicate code

```suggestion
public static isEnabled: boolean = true;
```